driver-async/src/main/com/mongodb/async/client/ClientSessionImpl.java

```diff
@@ -31,9 +31,15 @@
 
 class ClientSessionImpl extends BaseClientSessionImpl implements ClientSession {
 
+    private enum TransactionState {
+        NONE, IN, DONE, ABORTED
+    }
+
     private final OperationExecutor executor;
-    private boolean inTransaction;
+    private TransactionState transactionState = TransactionState.NONE;
     private boolean messageSent;
+    private boolean commitInProgress;
```

Comment:
This isn't pretty, but it didn't seem like a TransactionState so left it as a boolean.  Semantics are just that the value is true while a CommitTransaction operation is in progress and false otherwise.  This allows the session to report that it's in an active transaction when state is DONE only when a commit is in progress.